lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,436 @@
+    object_id: "0965570e-4e82-45ae-955a-c655dba6c014"
+    permissions: "Certificate Management"
+
+- name: Delete a key vault
+  azure_rm_keyvault:
+    resource_group: Testing
+    name: example.com
+    state: absent
+    location: westus
+
+'''
+
+RETURN = '''
+state:
+    description: Current state of the key vault.
+    returned: always
+    type: dict
+    example: {
+        "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/presentation_rg/providers/Microsoft.KeyVault/vaults/ozidatamanagement",
+        "location": "westus2",
+        "name": "keyvault",
+        "properties": {
+            "access_policies": [
+                {
+                    "application_id": null,
+                    "object_id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+                    "permissions": {
+                        "certificates": [],
+                        "keys": [
+                            "Get",
+                            "List",
+                            "UnwrapKey"
+                        ],
+                        "secrets": []
+                    },
+                    "tenant_id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+                }
+            ],
+            "enabled_for_deployment": true,
+            "enabled_for_disk_encryption": true,
+            "enabled_for_template_deployment": false,
+            "sku": {
+                "name": "premium"
+            },
+            "tenant_id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+            "vault_uri": "https:/keyvault.vault.azure.net"
+        },
+        "tags": {},
+        "type": "Microsoft.KeyVault/vaults"
+    }
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.keyvault.models import Vault, VaultCreateOrUpdateParameters, VaultProperties, AccessPolicyEntry, Sku, Permissions
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMKeyVault(AzureRMModuleBase):
+
+    def __init__(self):
+
+        # define user inputs from playbook
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(choices=['present', 'absent'], default='present', type='str'),
+            enabled_for_deployment=dict(default=False, type='bool'),
+            enabled_for_disk_encryption=dict(default=False, type='bool'),
+            enabled_for_template_deployment=dict(default=False, type='bool'),
+            location=dict(type='str'),
+            sku=dict(choices=['standard', 'premium'], default='standard', type='str'),
+            vault_uri=dict(type='str'),
+            tenant_id=dict(type='str'),
+            object_id=dict(type='str'),
+            application_id=dict(type='str'),
+            permissions=dict(choices=[
+                'Key, Secret, & Certificate Management',
```

Comment:
Is there an easy way to configure it? These string is too long and complicate (space/character case) to be a parameter choice.

Reply:
I'm open to suggestions, however I don't think how azure cli does it will translate any better to YAML. Sticking with this for now.

Reply:
I've added a comment on a better way of doing this in the documentation part for permissions. I agree this is not what we want to present to end users and is very much not like Ansible.